### PR TITLE
Add a shared code sync action

### PR DESCRIPTION
**Pull request by Tratcher**
This is a copy of the action used in AspNetCore that syncs shared code between runtime and aspnetcore. Another PR will go into dotnet/extensions with similar content to pull changes (manually) the other direction.

Sample:
- Run: https://github.com/Tratcher/dotnet-extensions-experimental/actions/runs/5084656947
- Issue: https://github.com/azure/dotnet-extensions-experimental/issues/1

Dotnet PR: https://github.com/dotnet/extensions/pull/4008

### CODE REVIEW

File: src/Shared/DotNetSync/ReadMe.SharedCode.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+The code in this directory is shared between dotnet/extensions and azure/dotnet-extensions-experimental. This contains shared sources used in both repos. Any changes to this dir need to be checked into both repositories.
```
Comment (RussKie) [inline src/Shared/DotNetSync/ReadMe.SharedCode.md:1]:
```suggestion
The code in this directory is shared between [dotnet/extensions](https://github.com/dotnet/extensions) and [azure/dotnet-extensions-experimental](https://github.com/Azure/dotnet-extensions-experimental). This contains shared sources used in both repos. Any changes to this directory need to be checked into both repositories.
```

File: src/Shared/DotNetSync/ReadMe.SharedCode.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+dotnet/extensions code paths:
+- extensions\src\shared\
```
Comment (RussKie) [inline src/Shared/DotNetSync/ReadMe.SharedCode.md:4]:
```suggestion
- [extensions\src\Shared\](https://github.com/dotnet/extensions/tree/main/src/Shared)
```

File: src/Shared/DotNetSync/ReadMe.SharedCode.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+azure/dotnet-extensions-experimental code paths:
+- dotnet-extensions-experimental\src\shared\
```
Comment (RussKie) [inline src/Shared/DotNetSync/ReadMe.SharedCode.md:7]:
```suggestion
- [dotnet-extensions-experimental\src\Shared\](https://github.com/Azure/dotnet-extensions-experimental/tree/main/src/Shared)
```

File: src/Shared/DotNetSync/ReadMe.SharedCode.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Copying code
+- To copy code from dotnet/extensions to azure/dotnet-extensions-experimental, set AZUREEXTENSIONS_REPO to the azure repo root and then run CopyToAzure.cmd.
+- To copy code from azure/dotnet-extensions-experimental to dotnet/extensions, set DOTNETEXTENSIONS_REPO to the dotnet repo root and then run CopyToDotnet.cmd.
```
Comment (RussKie) [inline src/Shared/DotNetSync/ReadMe.SharedCode.md:11]:
```suggestion
- To copy code from dotnet/extensions to azure/dotnet-extensions-experimental, set `AZUREEXTENSIONS_REPO` environment variable to the azure repo root and then run `CopyToAzure.cmd`.
- To copy code from azure/dotnet-extensions-experimental to dotnet/extensions, set `DOTNETEXTENSIONS_REPO` environment variable to the dotnet repo root and then run `CopyToDotnet.cmd`.
```

File: src/Shared/DotNetSync/ReadMe.SharedCode.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+## GitHub Actions
+
+In azure/dotnet-extensions-experimental, the [dotnet-sync](https://github.com/Azure/dotnet-extensions-experimental/actions/workflows/dotnet-sync.yml) GitHub action automatically creates PRs to pull in changes from dotnet/runtime.
```
Comment (RussKie) [inline src/Shared/DotNetSync/ReadMe.SharedCode.md:15]:
```suggestion
In azure/dotnet-extensions-experimental, the [dotnet-sync](https://github.com/Azure/dotnet-extensions-experimental/actions/workflows/dotnet-sync.yml) GitHub action automatically creates PRs to pull in changes from dotnet/extensions.
```

**Review by RussKie — APPROVED** [review]
👍 with few nits

**Comment by RussKie** [comment]
There's only one problem with sharing the code - license headers.
image

.NET code has the following license headers:
```
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
```

...and the Azure SDK has the following license header:
```
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
```

The Shared files are linked, meaning those effectively become a part of a project, which is MS-licensed.... We can't fix this unless we change the license headers in the shared files.

**Comment by Tratcher** [comment]
I'll follow up about the license headers.

**Pull request merged**